packages/mobile/src/components/lineup-tile/LineupTileAccessStatus.tsx

```diff
@@ -51,30 +55,44 @@ export const LineupTileAccessStatus = ({
   premiumConditions: PremiumConditions
 }) => {
   const styles = useStyles()
+  const dispatch = useDispatch()
   const isUSDCEnabled = useIsUSDCEnabled()
   const premiumTrackStatusMap = useSelector(getPremiumTrackStatusMap)
   const premiumTrackStatus = premiumTrackStatusMap[trackId]
   const staticWhite = useColor('staticWhite')
+  const isUSDCPurchase =
+    isUSDCEnabled && isPremiumContentUSDCPurchaseGated(premiumConditions)
+
+  const handlePurchasePress = useCallback(() => {
+    dispatch(
+      setVisibility({
+        drawer: 'PremiumTrackPurchase',
+        visible: true,
+        data: { trackId }
+      })
+    )
+  }, [dispatch, trackId])
 
   return (
-    <View
-      style={[
-        styles.root,
-        isUSDCEnabled && isPremiumContentUSDCPurchaseGated(premiumConditions)
-          ? styles.usdcPurchase
-          : null
-      ]}
+    <TouchableOpacity
+      onPress={isUSDCPurchase ? handlePurchasePress : undefined}
     >
-      {premiumTrackStatus === 'UNLOCKING' ? (
-        <LoadingSpinner style={styles.loadingSpinner} fill={staticWhite} />
-      ) : (
-        <IconLock fill={staticWhite} width={16} height={16} />
-      )}
-      <Text style={styles.text}>
-        {premiumTrackStatus === 'UNLOCKING'
-          ? messages.unlocking
-          : messages.locked}
-      </Text>
-    </View>
+      <View style={[styles.root, isUSDCPurchase ? styles.usdcPurchase : null]}>
+        {premiumTrackStatus === 'UNLOCKING' ? (
+          <LoadingSpinner style={styles.loadingSpinner} fill={staticWhite} />
+        ) : (
+          <IconLock fill={staticWhite} width={16} height={16} />
+        )}
+        <Text style={styles.text}>
+          {isUSDCPurchase
+            ? premiumTrackStatus === 'UNLOCKING'
+              ? null
+              : formatUSDCWeiToUSDString(premiumConditions.usdc_purchase.price)
```

Comment:
If I remember correctly, this format function does not add dollar signs, in case you need one here.